nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -58,3 +72,379 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
+            if desc["lang"] == "en":
+                cve_description = desc["value"]
+            else:
+                cve_description = "No description provided."
+        cve_published_date = cve_json.get("published")
+        cve_modified_date = cve_json.get("lastModified")
+        cve_impact = cve_json["metrics"]
+
+        # Determine URL
+        if len(cve_json["references"]) > 0:
+            cve_url = cve_json["references"][0].get("url", f"https://www.cvedetails.com/cve/{cve_name}/")
+        else:
+            cve_url = f"https://www.cvedetails.com/cve/{cve_name}/"
+
+        # Determine if V3 exists and set all params based on found version info
+        if cve_impact.get("cvssMetricV31"):
+            cvss_base_score = cve_impact["cvssMetricV31"][0]["cvssData"]["baseScore"]
+            cvss_severity = cve_impact["cvssMetricV31"][0]["cvssData"]["baseSeverity"]
+            if cve_impact.get("cvssMetricV2"):
+                cvssv2_score = cve_impact["cvssMetricV2"][0].get("exploitabilityScore", 10)
+            else:
+                cvssv2_score = 10
+            cvssv3_score = cve_impact["cvssMetricV31"][0].get("exploitabilityScore", 10)
+
+        elif cve_impact.get("cvssMetricV30"):
+            cvss_base_score = cve_impact["cvssMetricV30"][0]["cvssData"]["baseScore"]
+            cvss_severity = cve_impact["cvssMetricV30"][0]["cvssData"]["baseSeverity"]
+            if cve_impact.get("cvssMetricV2"):
+                cvssv2_score = cve_impact["cvssMetricV2"][0].get("exploitabilityScore", 10)
+            else:
+                cvssv2_score = 10
+            cvssv3_score = cve_impact["cvssMetricV30"][0].get("exploitabilityScore", 10)
+
+        else:
+            cvss_base_score = cve_impact["cvssMetricV2"][0]["cvssData"]["baseScore"]
+            cvss_severity = cve_impact["cvssMetricV2"][0]["baseSeverity"] or self.convert_v2_base_score_to_severity(
+                cvss_base_score
+            )
+            cvssv2_score = cve_impact["cvssMetricV2"][0].get("exploitabilityScore", 10)
+            cvssv3_score = 0
+
+        all_cve_info = {
+            "url": cve_url,
+            "description": cve_description,
+            "published_date": cve_published_date,
+            "modified_date": cve_modified_date,
+            "cvss_base_score": cvss_base_score,
+            "cvss_severity": cvss_severity,
+            "cvssv2_score": cvssv2_score,
+            "cvssv3_score": cvssv3_score,
+        }
+
+        return all_cve_info
+
+    def update_cve(self, current_dlc_cve: CVELCM, updated_cve: dict) -> None:
+        """Update CVE information if the latest info is newer than existing info.
+
+        Args:
+            current_dlc_cve (CVELCM): Current CVE object from the DLM database.
+            updated_cve (dict): Latest CVE information from the software pull.
+        """
+        try:
+            current_dlc_cve.description = (
+                f"{updated_cve['description'][0:251]}..."
```

Comment:
We have inconsistency here, anything over 255 chars gets truncated to 250 chars, but if something is between 251 and 255 chars, it won't get truncated.

Reply:
Updating to truncate to 255 chars (including ellipsis) if over 255 chars.  If it's less than that we'll take the entire description.  Looks like 255 is the max length for this description field.

Reply:
I don't believe the ellipses are adding anything, and this and the field should instead truncate to the `CHARFIELD_MAX_LENGTH` (which currently is `255`, but Nautobot's standard is to use the variable.

Reply:
I was only using the ellipsis to show that it's been truncated just in case the user is interested in looking up the full description if not shown.

I will update to use the constant.

Reply:
As we discussed, changed this over to a TextField so that the character count is irrelevant now.